### ISSUE TITLE
SpatialTransform with respect to different coordinate systems (Axes)

### ISSUE THREAD

**Issue by schuhschuh** [BioMedIA/deepali#52]
Currently, a `SpatialTransform` is a coordinate mapping of normalized coordinates with respect to the reference sampling `Grid` specified when constructing the spatial transformation object. A user of the transform has to map coordinates with respect to other `Axes` to these normalized coordinates before applying the transformation, and back again. In contexts where no normalized coordinates are used to represent the spatial transformation because `torch.nn.functional.grid_sample` is not being utilized, one might even want to parameterize the spatial transformation with respect to other coordinate systems (e.g., the world coordinate system). These use cases could be supported by adding a `axes: Axes` property and optionally a corresponding keyword argument to `SpatialTranform.forward(.., axes: Optional[Axes] = None)` which can be used to define the parameterization axes of the coordinate map and further to specify with respect to which axes the input points are defined.

Thanks @qiuhuaqi for this suggestion. Let's discuss here benefits or any reason why we wouldn't support this.

**Comment by aschuh-hf** [comment]
While this feature request is different, and may still be desirable for those who want to implement a classic registration which directly operates in world space rather than normalized point coordinates, a possible work-around is to use `SpatialTransform.points()` instead of having spatial transforms whose parameterization is with respect to unnormalized coordinates spaces.

See #62.

**Comment by qiuhuaqi** [comment]
sorry I must've missed the review request for #62. Yeah I can see this is a good workaround for my usecase, which is transforming the points which are in the un-normalised space (world space) with `SpatialTransform`.

**Comment by aschuh-hf** [comment]
Following up on this comment 2.5 years later...

> I think SpatialTransform.tensor() method right now can only produce fields in the normalised space? It could just be enough to just add axes as an argument to any relevant functions.

@qiuhuaqi For non-rigid transformations at least, you have also the option to use `SpatialTransform.flow().axes("world")` instead of `SpatialTransform.tensor()`. But indeed, there currently will be a conversion step involved between cube based coordinates and world coordinates.
